universe/Conditions.cpp

```diff
@@ -1477,7 +1501,11 @@ Homeworld::Homeworld() :
 Homeworld::Homeworld(std::vector<std::unique_ptr<ValueRef::ValueRef<std::string>>>&& names) :
     Condition(),
     m_names(std::move(names))
-{}
+{
+    m_root_candidate_invariant = boost::algorithm::all_of(m_names, [](auto& e){ return e->RootCandidateInvariant(); });
```

Comment:
can't use `SetConditionVariance` with `m_names`?

Reply:
This class was updated before I templated the `SetConditionVariance` function.

Reply:
I replaced `SetConditionVariance` with `boost::algorithm::all_of` calls.  When the new base class emerges the interface of `SetConditionVariance` will be unusable anyway, so better ditch it right now.